import-extension/src/test/kotlin/com/pkware/detekt/extensions/rules/staticImport/EnforceStaticImportTest.kt

```diff
@@ -0,0 +1,309 @@
+    private lateinit var env: KotlinCoreEnvironment
+
+    // We need to set up a custom KotlinCoreEnvironment and add the Truth and Arguments class paths to that environment,
+    // so they are available to generate the BindingContext for detekt rules under test with the
+    // EnforceStaticImportExtension.
+    @BeforeEach
+    fun setUp() {
+        wrapper = createEnvironment(
+            additionalRootPaths = listOf(
+                File(Truth::class.java.protectionDomain.codeSource.location.path),
+                File(Arguments::class.java.protectionDomain.codeSource.location.path)
+            )
+        )
+        env = wrapper.env
+    }
+
+    @AfterEach
+    fun tearDown() {
+        wrapper.dispose()
+    }
+
+    @Test
+    fun `rule report based on custom configuration`() {
+
+        val code = """
+            fun main() {
+                val value = 3.3
+                value = Math.floor(value)
+                print("3")
+            }
+            """
+
+        val findings = EnforceStaticImport(
+            TestConfig(mapOf(METHODS to listOf("java.lang.Math.floor", "kotlin.io.print")))
+        ).compileAndLintWithContext(env, code)
+
+        assertThat(findings).hasSize(1)
+        assertThat(findings).hasSourceLocations(
+            SourceLocation(3, 18),
+        )
+    }
+
+    @Test
+    fun `rule report truth and arguments`() {
+
+        val code = """
+            import com.google.common.truth.Truth
+            import com.google.common.truth.Truth.assertThat
+            import org.junit.jupiter.params.provider.Arguments
+            fun main() {
+                val value = 3
+                Truth.assertThat(value).isEqualTo(3)
+                assertThat(value).isNotEqualTo(4)
+                val args = Arguments.arguments("test", "test2")
+            }
+            """
+
+        val findings = EnforceStaticImport(
+            TestConfig(
+                mapOf(
+                    METHODS to listOf(
+                        "com.google.common.truth.Truth.assertThat",
+                        "org.junit.jupiter.params.provider.Arguments.arguments"
+                    )
+                )
+            )
+        ).compileAndLintWithContext(env, code)
+
+        assertThat(findings).hasSize(2)
+        assertThat(findings).hasSourceLocations(
+            SourceLocation(6, 11),
+            SourceLocation(8, 26)
+        )
+    }
+
+    @Test
+    fun `rule report with called methods using their fully qualified names`() {
+
+        val code = """
+            fun main() {
+                var value = 3.54
+                value = java.lang.Math.floor(value)
+            }
+            """
+
+        val findings = EnforceStaticImport(
+            TestConfig(mapOf(METHODS to listOf("java.lang.Math.floor")))
+        ).compileAndLintWithContext(env, code)
+
+        assertThat(findings).hasSize(1)
+        assertThat(findings).hasTextLocations(61 to 73)
```

Comment:
Is there any reason we're using `hasTextLocations` rather than continuing to use `hasSourceLocations`? Is it just to check that we're capturing the exact length of the fully qualified statement being flagged?

Reply:
umm that was the original spirit of hasTextLocations usage yes, the original tests were adapted straight from detekts own tests

Reply:
i am changing everything to hasSourceLocation